### DISCUSSION TITLE
Do you need to cork for sending websocket messages?

### DISCUSSION THREAD

**Discussion by chanon** [uNetworking/uWebSockets.js#1028] [Q&A] [answered]
I know that res.cork is recommended for best performance for http responses.
I am wondering about when sending websocket messages using [`WebSocket<UserData>.send()`](https://unetworking.github.io/uWebSockets.js/generated/interfaces/WebSocket.html#send.send-1) should `cork` be used too as I see there is a [`WebSocket<UserData>.cork()`](https://unetworking.github.io/uWebSockets.js/generated/interfaces/WebSocket.html#cork.cork-1) method.

Should I wrap every send in a cork?

Or is it helpful only for sending multiple messages (multiple sends) at once?

**Comment by uNetworkingAB** [comment, marked as answer]
Ideally every send should be wrapped in cork but if you don't, there are a few cases where it auto corks either way. So unless you can't easily fit in the lambda's preallocated capture memory, always cork. But you can also just benchmark your solution to find out.

**Reply by chanon** [reply]
Thank you for the quick and informative reply!
